cmd/ebrelayer/txs/relayToCosmos.go

```diff
@@ -3,50 +3,96 @@ package txs
 // DONTCOVER
 
 import (
-	"fmt"
+	"log"
+	"sync/atomic"
 
+	"github.com/Sifchain/sifnode/x/ethbridge"
+	"github.com/Sifchain/sifnode/x/ethbridge/types"
 	"github.com/cosmos/cosmos-sdk/client/context"
 	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/Sifchain/sifnode/x/ethbridge"
-	"github.com/Sifchain/sifnode/x/ethbridge/types"
+)
+
+var (
+	nextSequenceNumber uint64 = 0
 )
 
 // RelayToCosmos applies validator's signature to an EthBridgeClaim message containing
 // information about an event on the Ethereum blockchain before relaying to the Bridge
-func RelayToCosmos(cdc *codec.Codec, moniker, password string, claim *types.EthBridgeClaim, cliCtx context.CLIContext,
+func RelayToCosmos(cdc *codec.Codec, moniker, password string, claims []types.EthBridgeClaim, cliCtx context.CLIContext,
 	txBldr authtypes.TxBuilder) error {
+	var messages []sdk.Msg
+	log.Printf("ebrelayer RelayToCosmos with %d claims\n", len(claims))
+	log.Printf("ebrelayer RelayToCosmos nextSequenceNumber is %d\n", nextSequenceNumber)
 
-	msg := ethbridge.NewMsgCreateEthBridgeClaim(*claim)
+	for _, claim := range claims {
+		// Packages the claim as a Tendermint message
+		msg := ethbridge.NewMsgCreateEthBridgeClaim(claim)
 
-	err := msg.ValidateBasic()
-	if err != nil {
-		fmt.Println("failed to get message from claim with:", err.Error())
-		return err
+		err := msg.ValidateBasic()
+		if err != nil {
+			log.Println("failed to get message from claim with:", err.Error())
+			continue
+		} else {
+			messages = append(messages, msg)
+		}
 	}
 
 	// Prepare tx
-	txBldr, err = utils.PrepareTxBuilder(txBldr, cliCtx)
+	txBldr, err := utils.PrepareTxBuilder(txBldr, cliCtx)
 	if err != nil {
+		log.Println("error building tx: ", err)
+		log.Println("tx buidler response on error: ", txBldr)
 		return err
 	}
 
+	log.Printf("ebrelayer RelayToCosmos sequenceNumber is %d from tx builder\n", txBldr.Sequence())
+
+	// If we start to control sequence
+	if nextSequenceNumber > 0 {
+		txBldr.WithSequence(nextSequenceNumber)
+		log.Println("txBldr.WithSequence(nextSequenceNumber) passed")
+	}
+	
+	log.Println("building and signing")
 	// Build and sign the transaction
-	txBytes, err := txBldr.BuildAndSign(moniker, password, []sdk.Msg{msg})
+	txBytes, err := txBldr.BuildAndSign(moniker, password, messages)
 	if err != nil {
+		log.Printf("ebrelayer RelayToCosmos error 1 is %s\n", err.Error())
 		return err
 	}
-
+	
+	log.Println("built tx, now broadcasting")
 	// Broadcast to a Tendermint node
-	res, err := cliCtx.BroadcastTxSync(txBytes)
```

Comment:
I moved Async on suggestion from Tanmay. Even if we move back to sync, we should still wait because I think this broadcast will always resolve before the tx's are included in a block.